### DISCUSSION TITLE
Tooltip for disabled action button

### DISCUSSION THREAD

**Discussion by ManuDoni** [filamentphp/filament#4271] [Help]
Hello,
I would like to show a tooltip on a disabled action button to explain why that button is disabled.
Unfortunately, the tooltip is not displayed for disabled buttons.
Is there a way to do that? I could also make a PR but first I want to be sure if this behaviour is the intended one

**Comment by danharrin** [comment]
Sure, I think that tooltip for disabled buttons would be fine.

**Comment by ManuDoni** [comment]
I'm looking for the right place for this change.
It seems that the `alpine-tooltip` plugin shows the tooltip when some events are fired (by default `mouseenter` and `focus`) but disabled buttons don't fire events.
I tried using `pointerenter` event but it didn't work, furthermore, for enabled buttons, the tooltip won't disappear leaving the button with the mouse.
Maybe I'll have the time to dig into this problem in a few weeks.
Does @ryangjchandler know something about this issue?

**Reply by ryangjchandler** [reply]
@ManuDoni This isn't a thing that `alpine-tooltip` does, it's actually a thing that Tippy.js does (the under-underlying package powering the tooltips).

The way to work around it wrapping the button in another element and placing the tooltip there, since that element would still receive pointer events whilst the button is disabled.

We'd essentially need to move the tooltip attributes from the button itself to the next highest element, not sure how simple that is going to be and if there are any unwanted side effects.

**Comment by danny-eugene** [comment]
I've just run into the same issue, wondering if the move to v3 allows a simpler fix now?

Thanks

**Comment by emmadesilva** [comment]
Would also love this! It doesn't seem like Tippy.js is interested in implementing it, but the workarounds listed here seem relatively simple. @danharrin what do you think?

https://github.com/atomiks/tippyjs-react/issues/123
https://github.com/atomiks/tippyjs/issues/286

**Reply by danharrin** [reply]
For sure, you can open a PR to move the tooltip onto a parent element if you want. But please do not introduce new parent elements if they do not have a tooltip.

**Comment by bbredewold** [comment]
Is someone already doing this? If not, maybe I can make some time for this.

**Reply by danharrin** [reply]
No-one has made a PR that has been merged, no

**Comment by ericmp33** [comment]
Not sure if it's a great idea like this... I feel it kinda _hacky_. If you guys think it is fine, I can do a PR.
I also don't know if the wrapping tag should be a `div`, hmm maybe a `span`.
```js
// vendor/filament/actions/resources/views/components/action.blade.php

@props([
    'action',
    'dynamicComponent',
    'icon' => null,
])

@php
    $isDisabled = $action->isDisabled();
    $url = $action->getUrl();
    $tooltip = $action->getTooltip();
    $tooltipIsNotEmpty = str($tooltip)->isNotEmpty();
@endphp

@if ($tooltipIsNotEmpty) <div x-data x-tooltip.raw="{{ $tooltip }}"> @endif
    <x-dynamic-component
        :color="$action->getColor()"
        :component="$dynamicComponent"
        :disabled="$isDisabled"
        :form="$action->getFormToSubmit()"
        :form-id="$action->getFormId()"
        :href="$isDisabled ? null : $url"
        :icon="$icon ?? $action->getIcon()"
        :icon-size="$action->getIconSize()"
        :key-bindings="$action->getKeyBindings()"
        :label-sr-only="$action->isLabelHidden()"
        :tag="$url ? 'a' : 'button'"
        :target="($url && $action->shouldOpenUrlInNewTab()) ? '_blank' : null"
        :tooltip="$tooltip"
        :type="$action->canSubmitForm() ? 'submit' : 'button'"
        :wire:click="$action->getLivewireClickHandler()"
        :wire:target="$action->getLivewireTarget()"
        :x-on:click="$action->getAlpineClickHandler()"
        :attributes="
            \Filament\Support\prepare_inherited_attributes($attributes)
                ->merge($action->getExtraAttributes(), escape: false)
                ->class(['fi-ac-action'])
        "
    >
        {{ $slot }}
    </x-dynamic-component>
@if ($tooltipIsNotEmpty) </div> @endif
```

**Reply by danharrin** [reply]
Yeah that is the sort of solution we'd need

**Reply by ericmp33** [reply]
🫡 https://github.com/filamentphp/filament/pull/13671
